src/routes/index.js

```diff
@@ -245,6 +245,28 @@ router.get("/download", (req, res, next) => {
   return res.json({ message: "file no exist", success: false });
 });
 
+router.get("/delete", (req, res, next) => {
+  const fileHash = req.query.hash;
+  if (!fs.existsSync(mapPath)) {
+    return res.json({ message: "no files exist" });
+  }
+  let map = JSON.parse(fs.readFileSync(mapPath));
+  const fileDetails = map[fileHash].details;
+  let fileBackups = map[fileHash].backups;
+  let possiblePresentFile = fileBackups.filter(
+    file => file.hostname === process.env.SELF_HOSTNAME
+  );
+  if (possiblePresentFile.length !== 0) {
+    try {
+      fs.unlinkSync(possiblePresentFile[0].path + fileDetails.name);
```

Comment:
What we'd like to do, is delete the file clusterwide. So it would make sense if you can make async delete API calls to the rest of the servers and remove it from the map, and then store it.